auth/src/main/java/com/firebase/ui/auth/ui/accountlink/AccountLinker.java

```diff
@@ -0,0 +1,149 @@
+package com.firebase.ui.auth.ui.accountlink;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.ErrorCodes;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.ResultCodes;
+import com.firebase.ui.auth.ui.BaseHelper;
+import com.firebase.ui.auth.ui.TaskFailureLogger;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuthUserCollisionException;
+import com.google.firebase.auth.FirebaseUser;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class AccountLinker implements OnSuccessListener<AuthResult>, OnFailureListener {
```

Comment:
Welcome to the bad boy... I'll try my best to answer your questions, but even I struggle to understand what's going on! (It works though)